packages/utils/src/async/ordering.ts

```diff
@@ -0,0 +1,66 @@
+import { FIFO } from '../collection'
+
+type Item<T> = {
+  index: number
+  value?: T
+}
+export function ordered<T>() {
```

Comment:
is there a benefit of this over [heap](https://github.com/hoprnet/hoprnet/blob/01e945e13f1cf5f767cb95e373a364a4f08dfac2/packages/core-ethereum/src/indexer/index.ts#L2)?

Reply:
Yes, the problem is that events can occur more than once (and a heap takes way more computation).

Currently, the indexer subscribes to events such as:

`channel.on('*')` -> Announcemnet, ChannelUpdate etc.

In theory, the events arrive ordered and only once and using the heap there is some order guaranteed. But in reality, the indexer delivers events more than events as well as very late, which possibly makes the indexer handle events either not or more than once which produces concurrency issues such as write-after-write or write-after-read errors

Reply:
so the difference is that FIFO doesn't keep duplicated events where as HEAP does?

Reply:
Both, it does not keep them and it won't process them in a different order